src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp

```diff
@@ -112,20 +112,44 @@ NFTokenAcceptOffer::preclaim(PreclaimContext const& ctx)
         if ((*so)[sfAmount] > (*bo)[sfAmount])
             return tecINSUFFICIENT_PAYMENT;
 
-        // If the buyer specified a destination, that destination must be
-        // the seller or the broker.
+        // If the buyer specified a destination
         if (auto const dest = bo->at(~sfDestination))
         {
-            if (*dest != so->at(sfOwner) && *dest != ctx.tx[sfAccount])
+            // fixNFTokenBrokerAccept: Disabled
+            // the destination must be the seller or the broker.
+            if (!ctx.view.rules().enabled(fixNFTokenBrokerAccept) &&
+                *dest != so->at(sfOwner) && *dest != ctx.tx[sfAccount])
+            {
+                return tecNFTOKEN_BUY_SELL_MISMATCH;
+            }
+
+            // fixNFTokenBrokerAccept: Enabled
+            // the destination must be the tx account
+            if (ctx.view.rules().enabled(fixNFTokenBrokerAccept) &&
+                *dest != ctx.tx[sfAccount])
+            {
                 return tecNFTOKEN_BUY_SELL_MISMATCH;
```

Comment:
I personally would expect a `tecNO_PERMISSION` error code here, suggesting that the transaction submitting account has no permission to accept this offer. Similar when beeing in "non broker mode":

```
// If not in bridged mode...
        if (!so)
        {
            // If the offer has a Destination field, the acceptor must be the
            // Destination.
            if (auto const dest = bo->at(~sfDestination);
                dest.has_value() && *dest != ctx.tx[sfAccount])
                return tecNO_PERMISSION;
        }
```

The error code `tecNFTOKEN_BUY_SELL_MISMATCH` might be misleading.

Reply:
I agree. `tecNO_PERMISSION` is a better error response here. I will make this change.